### DISCUSSION TITLE
Absurd peak wattage and energy and inconsistency between current and wattage

### DISCUSSION THREAD

**Discussion by Isotech42** [zwave-js/zwave-js-ui#4060] [Q&A] [answered]
Hello,

I have a Z-Wave JS UI network running on a Raspberry Pi 3B+ (via Docker) with an Aeotec Z-Stick 7 attached via a USB extension lead and around thirty Aeotec SmartSwitch Gen5 plugs.

I periodically retrieve the plug values via the Z-Wave JS websocket to store the data in a database.

However, I've noticed that some plugs have recorded data that doesn't make sense, with energy rising from 0.335 kWh to **16,031 kWh** and then back to 0.34 kWh.
Or the power that goes from 5 Watt to peaks of **29,080W**, only to return to ‘normal’ wattage a few moments later.

As for the consistency between current and power, sometimes I get a current of 0.32A for 194W and another time I get 0.35A for 88W on the same plug.

So, does Z-Wave JS retrieve the values as they are (and the problem would therefore come from the plugs or the controller), or does it make calculations to display the values in the dashboard that can cause such errors?

Do you have any other ideas about where the problem might be coming from?

Thank you in advance for your answers!

**Comment by robertsLando** [comment]
I have a feel the issue is more likely with your device, Z-Wave JS doesn't do any manipulation of the values

cc @AlCalzone

**Reply by Isotech42** [reply]
Okay, was just to be sure. Thanks for your time!

**Comment by AlCalzone** [comment, marked as answer]
https://zwave-js.github.io/node-zwave-js/#/troubleshooting/nonsensical-values

**Reply by Isotech42** [reply]
The explanation is impressively detailed and easy to understand! Thank you so much!
(sorry for repeating a question that's already been answered though, I hadn't found this page before)
